### PR TITLE
👷 Add configuration to generate docs in CI

### PR DESCRIPTION
**Pull request by rosiemiller2**
### Description

Run the docs build once every time a branch is pushed to origin. Move those docs to the artifacts folder in Azure pipelines. The documentation files can then be downloaded and checked.

### Test Coverage

- [ ]  This change is covered by existing or additional automated tests.
- [x]  Manual testing has been performed (and evidence provided) as automated testing was not feasible.
- [ ]  Additional tests are not required for this change (e.g. documentation update).

### CODE REVIEW

File: azure-pipelines/build-release.yml [hunk 1]
```diff
@@ -13,32 +13,32 @@ stages:
         Linux_Py_3_6:
           python.version: '3.6'
           vmImageName: ubuntu-18.04
-          uploadCoverage: "false"
+          triggerNonMatrixActions: "false"
```
Comment (rosiemiller2) [inline azure-pipelines/build-release.yml]: Pretty terrible name - would welcome suggestions. Something that indicates that this is what triggers a step that we only want to run once, not on every machine in the matrix. :s

Comment (rosiemiller2) [reply azure-pipelines/build-release.yml]: Maybe this is the wrong way to do it and we should have action specific attributes. eg. 
```
uploadCoverage: "true"
generateDocs: "true"
```

Comment (muchzill4) [reply azure-pipelines/build-release.yml]: I like this approach (2 actions) better, it is more explicit and gives us more flexibility if we ever need it. I guess this is where I become a YAGNI hypocrite.

_Shame on me for suggesting single flag._

File: azure-pipelines/build-release.yml [hunk 2]
```diff
@@ -83,6 +89,15 @@ stages:
...
 
+    - script: |
+        python ci_scripts/generate_docs.py --output_dir $(System.DefaultWorkingDirectory)/docs_output
+      displayName: 'Generate documentation'
+      condition: and(succeeded(), eq(variables['generateDocs'], 'true'))
+
+    - publish: $(System.DefaultWorkingDirectory)/docs_output
+      artifact: Documentation
+      condition: and(succeeded(), eq(variables['generateDocs'], 'true'))
+
```
Comment (madchutney) [inline azure-pipelines/build-release.yml]: Would it make sense to do this in a separate stage? There is an overhead of installing the python dependencies but is this cleaner? The point of the `uploadCoverage` was that we needed the unit test results and didn't want to upload the results multiple times.

Comment (rosiemiller2) [reply azure-pipelines/build-release.yml]: You are right I think it should be. I will set it up. 👍

**Comment by rosiemiller2** [comment]
@madchutney This should be ready to go once it has finished building.

**Pull request merged**